playwright/_impl/_helper.py

```diff
@@ -362,3 +362,23 @@ def is_file_payload(value: Optional[Any]) -> bool:
         and "mimeType" in value
         and "buffer" in value
     )
+
+
+class BackgroundTaskTracker:
+    def __init__(self) -> None:
+        self._pending_tasks: List[asyncio.Task] = []
+
+    def create_task(self, coro: Coroutine) -> asyncio.Task:
```

Comment:
What's the advantage of creating it externally? We need to tightly control background tasks, so it seems safest to encapsulate all that's needed to do so like we have done here.

i.e. Don't use `asyncio.create_task`,—this will create leaks. Use `BackgroundTaskTracker.create_task` and we'll handle everything for you (like adding a done callback, adding it to pending tasks, and cancelling it eventually).

The thing that breaks us is calling `asyncio.create_task` and then not tracking it, so we make the call that you need to use to create the task, ensure it's being registered and cleaned up properly.